src/main/java/be/syntra/devshop/DevshopFront/services/CartServiceImpl.java

```diff
@@ -39,4 +43,55 @@ public void addToCart(Product product) {
         getCart().setProducts(productList);
         log.info("addToCart() -> {}", currentCart.getProducts().size());
     }
+
+    @Override
+    public StatusNotification payCart(CartDto cartDto) {
+        // cartDto is send to backend through rest template url with the users name which is the email.
+        return StatusNotification.SUCCESS;
+    }
+
+    @Override
+    public void cartTotalPrice(PaymentDto paymentDto) {
+        paymentDto.setTotalPrice(getCart().getProducts().stream()
+                .map(this::getTotalPerProduct)
+                .reduce(BigDecimal.ZERO, BigDecimal::add)
+        );
+    }
+
+    private BigDecimal getTotalPerProduct(Product product) {
+        return product.getPrice().multiply(new BigDecimal(product.getTotalInCart()));
+    }
+
+    @Override
+    public void addOneToProductInCart(Long productId) {
+        Product productToAlter = getProductById(productId);
+        // todo: DEV-034: update in cachedProducts-replacement
+        productToAlter.setTotalInCart(productToAlter.getTotalInCart() + 1);
+    }
+
+    @Override
+    public void removeOneFromProductInCart(Long productId) {
+        Product productToAlter = getProductById(productId);
+        // todo: DEV-034: update in cachedProducts-replacement
+        productToAlter.setTotalInCart(productToAlter.getTotalInCart() - 1);
+        if (productToAlter.getTotalInCart() == 0) {
+            // todo: DEV-034: update in cachedProducts-replacement
+            removeProductFromCart(productId);
+        }
+    }
+
+    @Override
+    public void removeProductFromCart(Long productId) {
+        Product productToRemove = getProductById(productId);
+        // todo: DEV-034: update in cachedProducts-replacement
+        productToRemove.setTotalInCart(0);
+        currentCart.getProducts().remove(productToRemove);
+    }
+
+    private Product getProductById(Long productId) {
+        return currentCart.getProducts().stream()
+                .filter(product -> product.getId() == productId)
```

Comment:
object comparison by .equals. You're streaming a CartDto, which has a List<Product>. The id is a Long. Only primitives and null are best compared with ==